weasyl/moderation.py

```diff
@@ -470,74 +470,57 @@ def setusermode(userid, form):
         note_about(userid, form.userid, 'User mode changed: action was %r' % (action,), message)
 
 
-def submissionsbyuser(userid, form):
-    if userid not in staff.MODS:
-        raise WeasylError("Unexpected")
-
+def submissionsbyuser(targetid):
     query = d.engine.execute("""
-        SELECT su.submitid, su.title, su.rating, su.unixtime, su.userid, pr.username, su.settings
+        SELECT su.submitid, su.title, su.rating, su.unixtime, su.settings
         FROM submission su
-            INNER JOIN profile pr USING (userid)
-        WHERE su.userid = (SELECT userid FROM login WHERE login_name = %(sysname)s)
+        WHERE su.userid = %(user)s
         ORDER BY su.submitid DESC
-    """, sysname=d.get_sysname(form.name))
+    """, user=targetid)
 
     ret = [{
         "contype": 10,
+        "userid": targetid,
         "submitid": i[0],
         "title": i[1],
         "rating": i[2],
         "unixtime": i[3],
-        "userid": i[4],
-        "username": i[5],
-        "settings": i[6],
+        "settings": i[4],
     } for i in query]
     media.populate_with_submission_media(ret)
     return ret
 
 
-def charactersbyuser(userid, form):
-    if userid not in staff.MODS:
-        raise WeasylError("Unexpected")
-
+def charactersbyuser(targetid):
     query = d.engine.execute("""
         SELECT
-            ch.charid, pr.username, ch.unixtime,
-            ch.char_name, ch.age, ch.gender, ch.height, ch.weight, ch.species,
-            ch.content, ch.rating, ch.settings, ch.page_views, pr.config
+            ch.charid, pr.username, ch.unixtime, ch.char_name, ch.rating,
+            ch.settings
         FROM character ch
-        INNER JOIN profile pr ON ch.userid = pr.userid
-        INNER JOIN login ON ch.userid = login.userid
-        WHERE login.login_name = %(sysname)s
-    """, sysname=d.get_sysname(form.name))
+            INNER JOIN profile USING (userid)
```

Comment:
```suggestion
            INNER JOIN profile pr USING (userid)
```

Looks like the ``pr`` got dropped. Trying to use ``/modcontrol/contentbyuser`` will fail as currently coded.

https://paste.weasyldev.com/show/4u1g4a2Vde6juT1CqYV0/
```
2020-02-21T01:00:32+0000 [stderr#error] ProgrammingError: (psycopg2cffi._impl.exceptions.ProgrammingError) missing FROM-clause entry for table "pr"
2020-02-21T01:00:32+0000 [stderr#error] LINE 3:             ch.charid, pr.username, ch.unixtime, ch.char_nam...
2020-02-21T01:00:32+0000 [stderr#error]                                ^
2020-02-21T01:00:32+0000 [stderr#error] [SQL:
2020-02-21T01:00:32+0000 [stderr#error]         SELECT
2020-02-21T01:00:32+0000 [stderr#error]             ch.charid, pr.username, ch.unixtime, ch.char_name, ch.rating,
2020-02-21T01:00:32+0000 [stderr#error]             ch.settings
2020-02-21T01:00:32+0000 [stderr#error]         FROM character ch
2020-02-21T01:00:32+0000 [stderr#error]             INNER JOIN profile USING (userid)
2020-02-21T01:00:32+0000 [stderr#error]         WHERE ch.userid = %(user)s
2020-02-21T01:00:32+0000 [stderr#error]     ]
```